src/commands/utils.rs

```diff
@@ -62,3 +71,93 @@ where
     let s = Vec::from_hex(&s).map_err(de::Error::custom)?;
     consensus::deserialize(&s).map_err(de::Error::custom)
 }
+
+/// Select coins for spend.
+///
+/// Returns the selected coins and the change amount, which could be zero.
+///
+/// `candidate_coins` are the coins to consider for selection.
+///
+/// `base_tx` is the transaction to select coins for. It should be without any inputs
+/// and without a change output, but with all non-change outputs added.
+///
+/// `change_txo` is the change output to add if needed (with any value).
+///
+/// `feerate_vb` is the minimum feerate (in sats/vb).
+///
+/// `min_fee` is the minimum fee (in sats) that the selection must have.
+///
+/// `max_sat_weight` is the maximum size difference (in vb) of
+/// an input in the transaction before and after satisfaction.
+pub fn select_coins_for_spend(
+    candidate_coins: &[CandidateCoin],
+    base_tx: bitcoin::Transaction,
+    change_txo: bitcoin::TxOut,
+    feerate_vb: u64,
+    min_fee: u64,
+    max_sat_weight: usize,
+) -> Result<(Vec<Coin>, bitcoin::Amount), InsufficientFunds> {
+    let max_input_weight = TXIN_BASE_WEIGHT + max_sat_weight as u32;
+    let candidates: Vec<Candidate> = candidate_coins
+        .iter()
+        .map(|cand| Candidate {
+            input_count: 1,
+            value: cand.coin.amount.to_sat(),
+            weight: max_input_weight,
+            is_segwit: true, // We only support receiving on Segwit scripts.
+        })
+        .collect();
+    // Transaction base weight is calculated from transaction with no inputs and no change output.
+    let base_weight = base_tx.weight().to_wu() as u32;
+    let long_term_feerate = FeeRate::from_sat_per_vb(LONG_TERM_FEERATE_VB as f32);
+    let target = Target {
+        value: base_tx.output.iter().map(|o| o.value).sum(),
+        feerate: FeeRate::from_sat_per_vb(feerate_vb as f32),
+        min_fee,
+    };
+    let drain_weights = DrainWeights {
+        output_weight: {
+            let mut tx_with_change = base_tx;
+            tx_with_change.output.push(change_txo);
+            tx_with_change.weight().to_wu() as u32 - base_weight
```

Comment:
I think the only unchecked conversion left after your changes in https://github.com/wizardsardine/liana/commit/dc533efe9b8c1db13503c9ded8938ebff1529052 is `feerate_vb as f32` where `feerate_vb` is `u64`. I could try converting it to `u16` and then `f32`, or is there a better way?

On a related note, do you think we should return an error if `feerate_vb` values greater than `MAX_FEERATE` are passed to `create_spend`?

Reply:
We could, but it's going to fail later anyways. IIRC i didn't add the explicit check to have the unit test which makes sure it does indeed fail the sanity check (the one which doesn't fail anymore with the correction of the round-up).

Reply:
I've applied your changes from https://github.com/wizardsardine/liana/commit/dc533efe9b8c1db13503c9ded8938ebff1529052 and also use `try_into` when converting `feerate_vb` to `u16` and then `f32`.

Reply:
We should make sure we don't crash if they provide a feerate value larger than 65k. But i don't think fixing this in this specific case makes a lot of sense, we should fix it consistently in all RPC commands at once so i've opened https://github.com/wizardsardine/liana/issues/800 to address this.

Reply:
In addition to https://github.com/wizardsardine/liana/issues/800, it might be easier to change the `select_coins_for_spend` parameter to `feerate_vb: f32` and do the type conversion within `create_spend` itself. Otherwise, I think the return error type of `select_coins_for_spend` will need to change later on.

Reply:
Indeed.